hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -632,9 +635,21 @@ public static void closeAll() throws IOException {
    */
   public static void closeAllForUGI(UserGroupInformation ugi)
   throws IOException {
+    if (LOGGER.isDebugEnabled()) {
+      debugLogFileSystemClose("closeAllForUGI", "UGI: " + ugi.toString());
+    }
     CACHE.closeAll(ugi);
   }
 
+  private static void debugLogFileSystemClose(String methodName, String additionalInfo) {
+    StackTraceElement callingMethod = new Throwable().fillInStackTrace().getStackTrace()[2];
+    LOGGER.debug(
+        "FileSystem." + methodName + "() called by method: "
```

Comment:
Prefer SLF4J {} and the values as varargs.
You might want to think that at TRACE the entire throwable is logged. Why so? one level up isn't always enough

Reply:
Hi Steve, thanks for taking a look at this!
Could you clarify what you mean by this?: You might want to think that at TRACE the entire throwable is logged. Why so? one level up isn't always enough